### DISCUSSION TITLE
canvas.setBackgroundImage from base64 does not appear

### DISCUSSION THREAD

**Discussion by yamatack** [fabricjs/fabric.js#8048] [Q&A]
this code triggered to button click does not appear background image,,,
but click second time appear background image, whats wrong with mycode.
this application can not use image-url for security.

```
$.ajax({
  type:"POST",
  url: "/path/to",
  data: postdata,
}).done(function(data, textStatus, jqXHR){
  bgImageData = data.src; /* "data:image/png;base64, ......" */
  imgObj = new Image();
  imgObj.src = bgImageData;
  canvasBgImg = new fabric.Image(imgObj);
  canvas.setBackgroundImage(canvasBgImg, canvas.renderAll.bind(canvas), {
    backgroundImageOpacity: 0.5,
    backgroundImageStretch: false
  });
  canvas.renderAll();
}) ....
```

**Comment by ShaMan123** [comment]
You should wait for the image to load before rendering the canvas

**Reply by yamatack** [reply]
[ShaMan123](https://github.com/ShaMan123)
thank you,
Ajax has already got return value(base64) ,and pathed to canvas object, but canvas background-image rendering does not work...
plz hlp me

**Reply by ShaMan123** [reply]
Read my comment.
Did you try it????
It's not related to fabric.
Attach to the onload hook of the image
